NSFS | S3 | Versioning: HeadObject on tagged object in version-enabled bucket does not return x-amz-tagging-count

### Environment info
- NooBaa Version: **5.15.4**
- Platform: **CentOS 8, deployed using [https://github.com/IBM/StorageScaleVagrant](https://github.com/IBM/StorageScaleVagrant)**

### Actual behavior

When performing a HeadObject operation on an object with multiple versions that got tagged (without versionId), there is no `x-amz-tagging-count` entry in the HTTP response headers:

```
        response = client.head_object(Bucket=bucket_name, Key=key)
        assert response['ResponseMetadata']['HTTPStatusCode'] == 200
>       assert response['ResponseMetadata']['HTTPHeaders']['x-amz-tagging-count'] == str(count)
E       KeyError: 'x-amz-tagging-count'
```

### Expected behavior

The HeadObject HTTP response header returns `x-amz-tagging-count` with a value of 1.

### Steps to reproduce

Execute the following extension code for Ceph s3-tests:

```
    @pytest.mark.versioning
    def test_s3cever12():
        client = get_client()
        bucket_name = get_new_bucket(client=client)
    
        # The following operation uses another client instance -> following operations will fail
    #    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
        _ces_safe_check_configure_versioning_retry(client, bucket_name, "Enabled", "Enabled")
    
        key = "test"
        version_ids = []
        contents = []
        num_versions = 3
        (version_ids, contents) = create_multiple_versions(client, bucket_name, key, num_versions)
    
        response = client.list_object_versions(Bucket=bucket_name)
        assert len(response['Versions']) == 3
    
        count = 1
        tagset = _create_simple_tagset(count)
        client.put_object_tagging(Bucket=bucket_name, Key=key, Tagging=tagset)
    
        response = client.head_object(Bucket=bucket_name, Key=key)
        print(response)
        assert response['ResponseMetadata']['HTTPStatusCode'] == 200
        assert response['ResponseMetadata']['HTTPHeaders']['x-amz-tagging-count'] == str(count)
```

### More information - Screenshots / Logs / Other output

Complete response metadata:

```
{'ResponseMetadata': {'RequestId': 'm0xz10l6-42drw2-7nf', 'HostId': 'm0xz10l6-42drw2-7nf', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-request-id': 'm0xz10l6-42drw2-7nf', 'x-amz-id-2': 'm0xz10l6-42drw2-7nf', 'access-control-allow-origin': '*', 'access-control-allow-credentials': 'true', 'access-control-allow-methods': 'GET,POST,PUT,DELETE,OPTIONS', 'access-control-allow-headers': 'Content-Type,Content-MD5,Authorization,X-Amz-User-Agent,X-Amz-Date,ETag,X-Amz-Content-Sha256,amz-sdk-invocation-id,amz-sdk-request', 'access-control-expose-headers': 'ETag,X-Amz-Version-Id', 'etag': '"mtime-d43j9qlyz9j4-ino-f0j"', 'last-modified': 'Wed, 11 Sep 2024 14:43:47 GMT', 'content-type': 'application/octet-stream', 'content-length': '9', 'accept-ranges': 'bytes', 'x-amz-version-id': 'mtime-d43j9qlyz9j4-ino-f0j', 'date': 'Wed, 11 Sep 2024 14:43:47 GMT', 'connection': 'keep-alive', 'keep-alive': 'timeout=5'}, 'RetryAttempts': 0}, 'AcceptRanges': 'bytes', 'LastModified': datetime.datetime(2024, 9, 11, 14, 43, 47, tzinfo=tzutc()), 'ContentLength': 9, 'ETag': '"mtime-d43j9qlyz9j4-ino-f0j"', 'VersionId': 'mtime-d43j9qlyz9j4-ino-f0j', 'ContentType': 'application/octet-stream', 'Metadata': {}}
```

Logs with "all" log level:
[noobaa_20240911_3.log.gz](https://github.com/user-attachments/files/16965417/noobaa_20240911_3.log.gz)

--- a/src/sdk/namespace_fs.js
+++ b/src/sdk/namespace_fs.js
@@ -52,7 +52,6 @@ const multi_buffer_pool = new buffer_utils.MultiSizeBuffersPool({
 
 const XATTR_USER_PREFIX = 'user.';
 const XATTR_NOOBAA_INTERNAL_PREFIX = XATTR_USER_PREFIX + 'noobaa.';
-const XATTR_NOOBAA_CUSTOM_PREFIX = XATTR_NOOBAA_INTERNAL_PREFIX + 'tag.';
 // TODO: In order to verify validity add content_md5_mtime as well
 const XATTR_MD5_KEY = XATTR_USER_PREFIX + 'content_md5';
 const XATTR_CONTENT_TYPE = XATTR_NOOBAA_INTERNAL_PREFIX + 'content_type';
@@ -63,6 +62,7 @@ const XATTR_VERSION_ID = XATTR_NOOBAA_INTERNAL_PREFIX + 'version_id';
 const XATTR_PREV_VERSION_ID = XATTR_NOOBAA_INTERNAL_PREFIX + 'prev_version_id';
 const XATTR_DELETE_MARKER = XATTR_NOOBAA_INTERNAL_PREFIX + 'delete_marker';
 const XATTR_DIR_CONTENT = XATTR_NOOBAA_INTERNAL_PREFIX + 'dir_content';
+const XATTR_TAG = XATTR_NOOBAA_INTERNAL_PREFIX + 'tag.';
 const HIDDEN_VERSIONS_PATH = '.versions';
 const NULL_VERSION_ID = 'null';
 const NULL_VERSION_SUFFIX = '_' + NULL_VERSION_ID;
@@ -1967,9 +1967,9 @@ class NamespaceFS {
             const stat = await file.stat(fs_context);
             if (stat.xattr) {
                 for (const [xattr_key, xattr_value] of Object.entries(stat.xattr)) {
-                    if (xattr_key.includes(XATTR_NOOBAA_CUSTOM_PREFIX)) {
+                    if (xattr_key.includes(XATTR_TAG)) {
                         tag_set.push({
-                            key: xattr_key.replace(XATTR_NOOBAA_CUSTOM_PREFIX, ''),
+                            key: xattr_key.replace(XATTR_TAG, ''),
                             value: xattr_value,
                         });
                     }
@@ -1993,7 +1993,7 @@ class NamespaceFS {
         }
         const fs_context = this.prepare_fs_context(object_sdk);
         try {
-            await this._clear_user_xattr(fs_context, file_path, XATTR_NOOBAA_CUSTOM_PREFIX);
+            await this._clear_user_xattr(fs_context, file_path, XATTR_TAG);
         } catch (err) {
             dbg.error(`NamespaceFS.delete_object_tagging: failed in dir ${file_path} with error: `, err);
             throw native_fs_utils.translate_error_codes(err, native_fs_utils.entity_enum.OBJECT);
@@ -2005,7 +2005,7 @@ class NamespaceFS {
         const fs_xattr = {};
         const tagging = params.tagging && Object.fromEntries(params.tagging.map(tag => ([tag.key, tag.value])));
         for (const [xattr_key, xattr_value] of Object.entries(tagging)) {
-              fs_xattr[XATTR_NOOBAA_CUSTOM_PREFIX + xattr_key] = xattr_value;
+              fs_xattr[XATTR_TAG + xattr_key] = xattr_value;
         }
         let file_path;
         if (params.version_id && this._is_versioning_enabled()) {
@@ -2017,7 +2017,7 @@ class NamespaceFS {
         dbg.log0('NamespaceFS.put_object_tagging: fs_xattr ', fs_xattr, 'file_path :', file_path);
         try {
             // remove existng tag before putting new tags
-            await this._clear_user_xattr(fs_context, file_path, XATTR_NOOBAA_CUSTOM_PREFIX);
+            await this._clear_user_xattr(fs_context, file_path, XATTR_TAG);
             await this.set_fs_xattr_op(fs_context, file_path, fs_xattr, undefined);
         } catch (err) {
             dbg.error(`NamespaceFS.put_object_tagging: failed in dir ${file_path} with error: `, err);
@@ -2316,6 +2316,10 @@ class NamespaceFS {
         return xattr[XATTR_MD5_KEY];
     }
 
+    _number_of_tags_fs_xttr(xattr) {
+        return Object.keys(xattr).filter(xattr_key => xattr_key.includes(XATTR_TAG)).length;
+    }
+
     /**
      * @param {string} bucket
      * @param {string} key
@@ -2336,6 +2340,7 @@ class NamespaceFS {
             mime.getType(key) || 'application/octet-stream';
         const storage_class = s3_utils.parse_storage_class(stat.xattr?.[XATTR_STORAGE_CLASS_KEY]);
         const size = Number(stat.xattr?.[XATTR_DIR_CONTENT] || stat.size);
+        const tag_count = stat.xattr ? this._number_of_tags_fs_xttr(stat.xattr) : 0;
 
         return {
             obj_id: etag,
@@ -2352,9 +2357,9 @@ class NamespaceFS {
             storage_class,
             restore_status: GlacierBackend.get_restore_status(stat.xattr, new Date(), this._get_file_path({key})),
             xattr: to_xattr(stat.xattr),
+            tag_count,
 
             // temp:
-            tag_count: 0,
             lock_settings: undefined,
             md5_b64: undefined,
             num_parts: undefined,

--- a/src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_pending_list.txt
+++ b/src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_pending_list.txt
@@ -177,7 +177,6 @@ s3tests_boto3/functional/test_s3.py::test_bucketv2_policy
 s3tests_boto3/functional/test_s3.py::test_bucket_policy_another_bucket
 s3tests_boto3/functional/test_s3.py::test_bucketv2_policy_another_bucket
 s3tests_boto3/functional/test_s3.py::test_get_obj_tagging
-s3tests_boto3/functional/test_s3.py::test_get_obj_head_tagging
 s3tests_boto3/functional/test_s3.py::test_put_max_tags
 s3tests_boto3/functional/test_s3.py::test_bucket_policy_put_obj_s3_noenc
 s3tests_boto3/functional/test_s3.py::test_copy_object_ifmatch_failed

